core-addon/Core.js

```diff
@@ -30,13 +30,13 @@ module.exports = class Core {
 
     // Asynchronously get the available studies. We don't need to wait
     // for this to finish, the UI can handle the wait.
-    this._availableStudies =
-      this._fetchAvailableStudies()
-          .then(studies => this._updateInstalledStudies(studies));
-            // Register a listener to react to remote-settings updates.
-            browser.firefoxPrivilegedApi.onRemoteSettingsSync.addListener((result) => {
-              this._sendStateUpdateToUI(result[0].data);
-          });
+    this._availableStudies = this._fetchAvailableStudies().then((studies) =>
+      this._updateInstalledStudies(studies)
+    );
+    // Register a listener to react to remote-settings updates.
+    browser.firefoxPrivilegedApi.onRemoteSettingsSync.addListener((result) => {
+      this._sendStateUpdateToUI(result[0].data);
+    });
```

Comment:
This is not working because `_sendStateUpdateToUI` doesn't take an argument. To make responding to changes work, you actually need to do something like this:

```suggestion
    browser.firefoxPrivilegedApi.onRemoteSettingsSync.addListener(async function (result) {
            // Important: this may be racing with the initial update or if there's two consecutive updates?
            this._availableStudies = await this._updateInstalledStudies(result[0].data);
            this._sendStateUpdateToUI();
    });
```

Reply:
aha, thank you! I figured I was just doing something wrong.

Reply:
(well I can't take this suggestion as-is since this is a constructor so we can't `await` but I understand what you mean :) )

Reply:
Ugh OK well you are using an `async` function for the callback, nevermind! `await` sounds great.